src/components/input/input.spec.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  fakeAsync,
+  inject,
+  ComponentFixture,
+  TestComponentBuilder,
+  injectAsync,
+  tick,
+} from 'angular2/testing';
+
+import {
+  it,
+  expect,
+  beforeEach,
+} from '../../core/facade/testing';
+
+import {Component} from 'angular2/core';
+import {By} from 'angular2/platform/browser';
+
+import {
+  MdInput, MdInputDuplicatedHintException, MD_INPUT_DIRECTIVES,
+  MdInputPlaceholderConflictException, MdInputUnsupportedTypeException
+} from './input';
+
+
+export function main() {
+  describe('MdInput', function () {
+    var builder: TestComponentBuilder;
+
+    beforeEach(inject([TestComponentBuilder], function (tcb: TestComponentBuilder) {
+      builder = tcb;
+    }));
+
+    it('creates a native <input> element', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          fixture.detectChanges();
+          expect(fixture.debugElement.query(By.css('input'))).not.toBeNull();
```

Comment:
`getAttribute()` returns `null` or  `""`, never `undefined` according to spec.

But yes, that would pass since `toBe` is strict equality. Replaced with `toBeTruthy()`.